src/Logika/LogikaHook.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState, createContext } from "react";
+
+const useDonationLogic = (initialDonations, initialDonations2) => {
+
+    const [donations, setDonations] = useState(0);
+    const [donations2, setDonations2] = useState(0);
+
+    const [totalDonate, setTotalDonate] = useState(false);
+    const [totalDonate2, setTotalDonate2] = useState(false);
+
+    const [progress, setProgress] = useState(0)
+    const [progress2, setProgress2] = useState(0)
+
+
+    const handleClick = (donate) => {
+        setDonations((prev) => prev + donate)
+    }
+    const handleClick2 = (donate2) => {
+        setDonations2((prev) => prev + donate2)
+    }
+    useEffect(() => {
+        setProgress((donations / initialDonations) * 100)
```

Comment:
tutaj jest to nieco przekombinowane, domyślam się że chodzi o obliczenie szerokości paska postępu?